### DISCUSSION TITLE
How to execute tests conditionally (only on given environment)

### DISCUSSION THREAD

**Discussion by bartoszWesolowski** [cypress-io/cypress#14552] [Questions and Help] [answered]
Hello,

I am writing tests that check out whether one of our integrations works correctly. Unfortunately this integration can not be tested on local env. As I want to have all tests passing on each env I would like to run this test only on selected envs (each but local). 

Is there a clean way to do this ? Currently i have test with if statement like:

```
it("test integration", () => {
   if (!isLocal()) {
   // run the actual test
   }
})
```

Is there a better way to do this ? I did a bit of googling and I found https://github.com/cypress-io/cypress-skip-test which seems to resolve my issue but it's also similar to the `if` approach.

I was wondering whether there is something more like config described here [Suite-configuration](https://docs.cypress.io/guides/core-concepts/writing-and-organizing-tests.html#Suite-configuration), where along with `browser: '!chrome'` I could add `env: '!local'` ?

```
describe('When NOT in Chrome', { env: '!local' }, () => {
  it('Shows warning', () => {
     // test only on local
  })
})
```
Thanks for your help :)

**Comment by bahmutov** [comment, marked as answer]
We suggest using `cypress-skip-test`, there are no current plans to introduce filtering like this into the core test runner

**Reply by bartoszWesolowski** [reply]
Ok, thanks for the update
